crates/transpiler/src/passes/consolidate_blocks.rs

```diff
@@ -44,6 +52,88 @@ pub enum DecomposerType {
     TwoQubitControlledU(TwoQubitControlledUDecomposer),
 }
 
+fn get_matrix(gate: &StandardGate) -> ArrayView2<Complex64> {
+    match gate {
+        StandardGate::CX => aview2(&CX_GATE),
+        StandardGate::CY => aview2(&CY_GATE),
+        StandardGate::CZ => aview2(&CZ_GATE),
+        StandardGate::CH => aview2(&CH_GATE),
+        StandardGate::DCX => aview2(&DCX_GATE),
+        StandardGate::ISwap => aview2(&ISWAP_GATE),
+        StandardGate::ECR => aview2(&ECR_GATE),
+        _ => unreachable!("Unsupported gate"),
+    }
+}
+
+/// Helper function that extracts the decomposer and basis gate directly from the [Target].
+#[inline]
+fn get_decomposer_and_basis_gate(
+    target: Option<&Target>,
+    approximation_degree: f64,
+) -> (DecomposerType, StandardGate) {
+    if let Some(target) = target {
+        // Targets from C should only support Standard gates.
+        if let Some(gate) = target.operations().find_map(|op| {
+            op.operation
+                .try_standard_gate()
+                .and_then(|gate| matches!(gate, PARAM_SET!()).then_some(gate))
+        }) {
+            return (
+                DecomposerType::TwoQubitControlledU(
+                    TwoQubitControlledUDecomposer::new(RXXEquivalent::Standard(gate), "ZXZ")
+                        .unwrap_or_else(|_| {
+                            panic!(
+                                "Error while creating Controlled U decomposer using a {} gate.",
+                                gate.name()
+                            )
+                        }),
+                ),
+                gate,
+            );
+        }
+        if let Some(gate) = target.operations().find_map(|op| {
+            op.operation
+                .try_standard_gate()
+                .and_then(|gate| matches!(gate, TWO_QUBIT_BASIS_SET!()).then_some(gate))
+        }) {
+            return (
+                DecomposerType::TwoQubitBasis(
+                    TwoQubitBasisDecomposer::new_inner(
+                        gate.into(),
+                        SmallVec::default(),
+                        get_matrix(&gate),
+                        approximation_degree,
+                        "U",
+                        None,
+                    )
+                    .unwrap_or_else(|_| {
+                        panic!(
+                            "Error while creating Basis Decomposer using a {} gate.",
+                            gate.name()
+                        )
+                    }),
+                ),
+                gate,
+            );
+        }
+    }
+    let gate = StandardGate::CX;
+    (
+        DecomposerType::TwoQubitBasis(
+            TwoQubitBasisDecomposer::new_inner(
+                gate.into(),
+                SmallVec::default(),
+                aview2(&CX_GATE),
+                1.0,
+                "U",
+                None,
+            )
+            .expect("Error while creating Basis Decomposer using a 'cx' gate."),
```

Comment:
```suggestion
            .expect("Error while creating TwoQubitBasisDecomposer using a 'cx' gate."),
```

Reply:
Addressed in e08cd6b91ad5614bddf77ec94ccf26fef8daddd5